stdlib/3/subprocess.pyi

```diff
@@ -43,12 +43,36 @@ if sys.version_info >= (3, 5):
                      stderr: Optional[_TXT] = ...) -> None: ...
         def check_returncode(self) -> None: ...
 
-    if sys.version_info >= (3, 6):
-        # Nearly same args as Popen.__init__ except for timeout, input, and check
+    if sys.version_info >= (3, 7):
+        # Nearly the same args as for 3.6, except for capture_output and text
         def run(args: _CMD,
-                timeout: Optional[float] = ...,
-                input: Optional[_TXT] = ...,
+                bufsize: int = ...,
+                executable: _PATH = ...,
+                stdin: _FILE = ...,
+                stdout: _FILE = ...,
+                stderr: _FILE = ...,
+                preexec_fn: Callable[[], Any] = ...,
+                close_fds: bool = ...,
+                shell: bool = ...,
+                cwd: Optional[_PATH] = ...,
+                env: Optional[_ENV] = ...,
+                universal_newlines: bool = ...,
+                startupinfo: Any = ...,
+                creationflags: int = ...,
+                restore_signals: bool = ...,
+                start_new_session: bool = ...,
+                pass_fds: Any = ...,
+                *,
+                capture_output: bool = ...,
                 check: bool = ...,
+                encoding: Optional[str] = ...,
+                errors: Optional[str] = ...,
+                input: Optional[_TXT] = ...,
+                text: Optional[str] = ...,
```

Comment:
I think it's actually `Optional[bool]`. It's meant as an alias for `extended_newlines`, and the `None` default is there so it can guard against both arguments being passed.

Reply:
Oh d'oh, I see what you mean. The '(if set)' gives it away: it's `None`, `False`, or `True`, and if the latter will use the separate `encoding` arg, `or` whatever default. Unclear to me why it's `Optional[bool]` not just `bool`, but hey-ho.